src/platform/packages/shared/kbn-esql-ast/src/commands_registry/registry.ts

```diff
@@ -79,6 +79,11 @@ export interface ICommandMetadata {
   types?: Array<{ name: string; description: string }>; // Optional property for command-specific types
   license?: LicenseType; // Optional property indicating the license for the command's availability
   observabilityTier?: string; // Optional property indicating the observability tier availability
+  subqueryRestrictions?: {
+    allowedInside: boolean; // Command is allowed inside subqueries
+    allowedOutside: boolean; // Command is allowed outside subqueries (at root level)
+  };
+  isSubqueryEntryPoint?: boolean; // Indicates command can start a subquery (e.g., FROM, ROW, SHOW)
```

Comment:
If I say which command can be an entry point for subquery I avoid hardcode and it is harmless, maybe in the future other source commands will be supported too

Reply:
If in the future we support the rest source commands I prefer to not have this flag at all. At this point it makes sense